compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -858,6 +869,54 @@ public void visit(BLangSimpleVariable varNode) {
         transferForkFlag(varNode);
     }
 
+    private void analyzeTypeNode(BLangType typeNode, SymbolEnv env) {
+        if (typeNode.getKind() == NodeKind.USER_DEFINED_TYPE) {
+            return;
+        }
+        if (typeNode.getKind() == NodeKind.RECORD_TYPE && !((BLangRecordTypeNode) typeNode).analyzed) {
+            analyzeDef(typeNode, env);
```

Comment:
Shouldn't we return here?

If `typeNode.getKind() == NodeKind.RECORD_TYPE` is true and `((BLangRecordTypeNode) typeNode).analyzed` is also true, we can still return here, right?